src/stores/FlairStore.js

```diff
@@ -149,15 +152,22 @@ class FlairStore {
             return this._groupProfiles[groupId];
         }
 
-        const profile = await matrixClient.getGroupProfile(groupId);
+        // No request yet, start one
+        if (!this._groupProfilesPromise[groupId]) {
+            this._groupProfilesPromise[groupId] = matrixClient.getGroupProfile(groupId);
+        }
+
+        const profile = await this._groupProfilesPromise[groupId];
         this._groupProfiles[groupId] = {
             groupId,
             avatarUrl: profile.avatar_url,
             name: profile.name,
             shortDescription: profile.short_description,
         };
+
         setTimeout(() => {
             delete this._groupProfiles[groupId];
+            delete this._groupProfilesPromise[groupId];
```

Comment:
why does this have to happen in the callback? can't it happen as soon as the promise returns?

Reply:
You would think that - I did. Empirically that didn't work because it means that a promise culd be deleted whilst something is still awaiting it. It's a race because in one version of the resolution, the promise would be deleted. 

In this new version I think I've inadvertently created a weird barrier where every resolution sets `this._groupProfiles[groupId]` and sets its own timeout. Maybe it shouldn't do that.....

Reply:
I don't see why it's a problem if `this._groupProfilesPromise` is cleared while something is waiting for the promise.

Reply:
Actually my last comment doesn't make sense because the `await` already has a reference to the promise, it doesn't matter if a reference to the same promise is remove from `this._groupProfilesPromise`.